ansible-doc failing on specific line in documentation

<!--- Verify first that your issue is not already reported on GitHub -->
<!--- Also test if the latest release and devel branch are affected too -->
<!--- Complete *all* sections as described, this form is processed automatically -->

##### SUMMARY
<!--- Explain the problem briefly below -->
anisble-doc is failing on below line
https://github.com/dynatrace-innovationlab/ansible_dynatrace_problem_comment/blob/master/library/dynatrace_comment.py#L47
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below, use your best guess if unsure -->
ansible-doc
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below

```

##### CONFIGURATION
<!--- Paste verbatim output from "ansible-config dump --only-changed" between quotes -->
```paste below

```

##### OS / ENVIRONMENT
<!--- Provide all relevant information below, e.g. target OS versions, network device firmware, etc. -->


##### STEPS TO REPRODUCE
<!--- Describe exactly how to reproduce the problem, using a minimal test-case -->
ansible-doc changes the line to something like below
u'Source where the comment originates from (default': u'Ansible)'
The above is not a valid python string and hence it throws an exception
<!--- Paste example playbooks or commands between quotes below -->
```yaml

```

<!--- HINT: You can paste gist.github.com links for larger files -->

##### EXPECTED RESULTS
<!--- Describe what you expected to happen when running the steps above -->
No exception  should be thrown, instead ansible-doc should fail the test

##### ACTUAL RESULTS
<!--- Describe what actually happened. If possible run with extra verbosity (-vvvv) -->
ansible-doc -t module dynatrace_innovationlab.dynatrace_collection.dynatrace_comment dynatrace_innovationlab.dynatrace_collection.dynatrace_deployment -vvv

<!--- Paste verbatim command output between quotes -->
```paste below
Traceback (most recent call last):
  File "/home/abehl/work/src/anshul_ansible/ansible/bin/ansible-doc", line 111, in <module>
    exit_code = cli.run()
  File "/home/abehl/work/src/anshul_ansible/ansible/lib/ansible/cli/doc.py", line 198, in run
    textret = DocCLI.format_plugin_doc(plugin, loader, plugin_type, search_paths)
  File "/home/abehl/work/src/anshul_ansible/ansible/lib/ansible/cli/doc.py", line 309, in format_plugin_doc
    text += DocCLI.get_man_text(doc)
  File "/home/abehl/work/src/anshul_ansible/ansible/lib/ansible/cli/doc.py", line 598, in get_man_text
    DocCLI.add_fields(text, doc.pop('options'), limit, opt_indent)
  File "/home/abehl/work/src/anshul_ansible/ansible/lib/ansible/cli/doc.py", line 476, in add_fields
    text.append(textwrap.fill(DocCLI.tty_ify(entry), limit, initial_indent=opt_indent, subsequent_indent=opt_indent))
  File "/home/abehl/work/src/anshul_ansible/ansible/lib/ansible/cli/__init__.py", line 426, in tty_ify
    t = cls._ITALIC.sub("`" + r"\1" + "'", text)    # I(word) => `word'
TypeError: expected string or buffer
```

Running this module through `validate-modules` produces the following error:

```
dynatrace_comment.py:0:0: E305 DOCUMENTATION.options.context.description.0: expected str @ data['options']['context']['description'][0]. Got {'Source where the comment originates from (default': 'Ansible)'}
```

This shows that pyyaml is parsing that yaml into a `key: value` pair, instead of a string due to the inclusion of the `:`

If you wrap that whole line in single quotes the problem will be resolved.

```
      - 'Source where the comment originates from (default: Ansible)'
```

We should probably catch that exception, and display some nicer error though.

--- a/changelogs/fragments/60587-doc_parsing.yml
+++ b/changelogs/fragments/60587-doc_parsing.yml
@@ -1,0 +1,2 @@
+bugfixes:
+  - Handle exception encountered while parsing the argument description in module when invoked via ansible-doc command (https://github.com/ansible/ansible/issues/60587).

--- a/lib/ansible/cli/doc.py
+++ b/lib/ansible/cli/doc.py
@@ -472,9 +472,13 @@ class DocCLI(CLI):
             text.append("%s %s" % (opt_leadin, o))
 
             if isinstance(opt['description'], list):
-                for entry in opt['description']:
+                for entry_idx, entry in enumerate(opt['description'], 1):
+                    if not isinstance(entry, string_types):
+                        raise AnsibleError("Expected string in description of %s at index %s, got %s" % (o, entry_idx, type(entry)))
                     text.append(textwrap.fill(DocCLI.tty_ify(entry), limit, initial_indent=opt_indent, subsequent_indent=opt_indent))
             else:
+                if not isinstance(opt['description'], string_types):
+                    raise AnsibleError("Expected string in description of %s, got %s" % (o, type(opt['description'])))
                 text.append(textwrap.fill(DocCLI.tty_ify(opt['description']), limit, initial_indent=opt_indent, subsequent_indent=opt_indent))
             del opt['description']
 

